File: 03-receive-message/receive.js
const express = require('express');
const app = express();
const PORT = 8000;

// This helps Express understand JSON data
app.use(express.json());

// This function receives WhatsApp messages from Openlyne
app.post('/webhook', (req, res) => {
    const payload = req.body;
    
    // Check if this is a new message
    if (payload.event === 'messages.upsert') {
        const data = payload.data || {};
        console.log(`📱 New message: ${JSON.stringify(data, null, 2)}`);
    }
    
    // Add your response logic here
    // For example: send a reply, save to database, etc.
    // Call the function send_text(data.key.remoteJid, "Hello from Openlyne!");

    res.json({ status: 'received' });
});

// Start the server
app.listen(PORT, '0.0.0.0', () => {
    console.log('🚀 Starting Openlyne webhook server...');
    console.log(`🔗 Webhook URL: http://localhost:${PORT}/webhook`);
});

module.exports = app;

File: 03-receive-message/receive.py
from fastapi import FastAPI
import uvicorn

app = FastAPI()

@app.post("/webhook")
async def receive_webhook(payload: dict):
    """
    This function receives WhatsApp messages from Openlyne
    """
    # Check if this is a new message
    if payload.get("event") == "messages.upsert":
        data = payload.get("data", {})
        print(f"📱 New message: {data}")
    
    # Add your response logic here
    # For example: send a reply, save to database, etc.

    return {"status": "received"}

if __name__ == "__main__":
    print("🚀 Starting openlyne webhook server...")
    print("🔗 Webhook URL: http://localhost:8000/webhook")
    uvicorn.run(app, host="0.0.0.0", port=8000)